src/main/java/org/opensearch/performanceanalyzer/jvm/ThreadList.java

```diff
@@ -138,8 +138,13 @@ public String toString() {
      * acquire this lock and move on if we could not get it.
      *
      * @return A hashmap of threadId to threadState.
+     * @param threadContentionMonitoringEnabled
      */
-    public static Map<Long, ThreadState> getNativeTidMap() {
+    public static Map<Long, ThreadState> getNativeTidMap(
+            boolean threadContentionMonitoringEnabled) {
+        if (threadBean.isThreadContentionMonitoringSupported()) {
+            threadBean.setThreadContentionMonitoringEnabled(threadContentionMonitoringEnabled);
+        }
```

Comment:
This check and set happens during every ```OSMetricsCollector#collectMetrics``` call which is very frequent. Do we want to set it once during initialization and when there is state change in ```threadContentionMonitoringEnabled```?

Reply:
This check is not very costly
If the flag is already enabled then this short-circuits out of the method without doing anything.
 ```
if (contentionMonitoringEnabled != enable) {
                if (enable) {
                    // if reeabled, reset contention time statistics
                    // for all threads
                    resetContentionTimes0(0);
                }

                // update the VM of the state change
                setThreadContentionMonitoringEnabled0(enable);

                contentionMonitoringEnabled = enable;
            }
```

Also, the OSMetricsCollector is initialized in the PA Plugin. We can't do a set once and update on state change sort of thing from there as ThreadList.getNativeTidMap is a static call and will need to know thread contention monitoring is enabled or not before every run.